CoreFoundation/URL.subproj/CFURL.c

```diff
@@ -4459,7 +4459,23 @@ CFStringRef CFURLCreateStringWithFileSystemPath(CFAllocatorRef allocator, CFURLR
     
     if ( relPath ) {
         // relPath is not absolute if it is zero length or doesn't start with a slash
-        Boolean relPathIsRelative = ((CFStringGetLength(relPath) != 0) ? (CFStringGetCharacterAtIndex(relPath, 0) != '/') : TRUE);
+        CFIndex length = CFStringGetLength(relPath);
+        Boolean relPathIsRelative = TRUE;
+        switch (fsType) {
+        case kCFURLPOSIXPathStyle:
+          relPathIsRelative = length > 0 && CFStringGetCharacterAtIndex(relPath, 0) != '/';
+          break;
+        case kCFURLHFSPathStyle:
+          relPathIsRelative = length > 0 && CFStringGetCharacterAtIndex(relPath, 0) == ':';
+          break;
+        case kCFURLWindowsPathStyle:
+          // This is not really ideal, but, since a drive letter a single
+          // character, and ':' is not a valid path character, and ':' is
+          // reserved for the ADS separator, this works.  In theory, on
+          // TARGET_DEPLOYMENT_WINDOWS we could use PathIsRelativeW here.
+          relPathIsRelative = length > 1 && CFStringGetCharacterAtIndex(relPath, 1) != ':';
```

Comment:
Per discussion: this needs to cover UNC paths, and then it’s good to go.

Reply:
Added a check for the UNC paths too.